hooks/useRequestDataWithPagination/index.mjs

```diff
@@ -0,0 +1,145 @@
+ *
+ * // Access pagination info
+ * console.log(`Showing page 2 of ${pagination.totalPages} (${pagination.totalItems} total items)`);
+ */
+export const useRequestDataWithPagination = (entity='postType', kind='post', query = {} ) => {
+	// State to hold pagination information
+	// This will hold total items and total pages.
+	const [pagination, setPagination] = useState({
+		totalItems: 0,
+		totalPages: 0,
+		perPage: query.per_page || 10, // Default to 10 items per page if not specified
+	});
+
+	// Use the existing useRequestData hook to fetch records
+	// This will return the records and loading state.
+	// The query can be an object or a number (for single record).
+	// If a number is passed, it will use getEntityRecord instead of getEntityRecords
+	// to fetch a single record.
+	// If an object is passed, it will use getEntityRecords to fetch multiple records.
+	const [ records, isLoading, invalidateResolver ] = useRequestData(entity, kind, query);
+
+	/**
+	 * Fetches the entity configuration for the specified entity and kind.
+	 * This allows us to construct the API endpoint for fetching pagination information.
+	 *
+	 * @effect
+	 * @dependency {string} entity
+	 * @dependency {string} kind
+	 * @returns {Object} The entity configuration object, or undefined if not found.
+	 */
+	const entityConfig = useSelect(
+		(select) => {
+			// Use getEntitiesByKind to get the entity config.
+			const entities = select(coreStore).getEntitiesByKind(entity);
+			return entities?.find( e => e.name === kind );
+		},
+		[entity, kind]
+	);
+
+	/**
+	 * Fetches pagination information from the WordPress REST API.
+	 *
+	 * This effect runs whenever records, entity, kind, query, or entityConfig changes.
+	 * It makes a direct API request to the same endpoint that getEntityRecords uses,
+	 * but with a minimal per_page setting to reduce data transfer.
+	 *
+	 * The effect extracts total items and total pages from the response headers
+	 * (X-WP-Total and X-WP-TotalPages) and updates the pagination state.
+	 *
+	 * ToDo: In future WordPress versions (6.4+), this might be replaceable with
+	 * the getEntityRecordsTotalItems and getEntityRecordsTotalPages selectors.
+	 *
+	 * @effect
+	 * @dependency {Array} [records, entity, kind, JSON.stringify(query), entityConfig, pagination]
+	 */
+	useEffect(() => {
+		if ( entityConfig ) {
+			// Construct the same API path that getEntityRecords uses.
+			const path = addQueryArgs( entityConfig.baseURL, {
+				...entityConfig.baseURLParams,
+				...query,
+				// Request the same number of records per page as specified in the query,
+				// or default to 10 if not specified.
+				per_page: pagination.perPage,
+				page: 1, // Only request the first page to get total items and pages.
+			});
+
+			// Make a direct fetch to the REST API.
+			apiFetch( {
+				path,
+				parse: false
+			}).then( response => {
```

Comment:
I would use an async keyword rather than .then here, like this:

https://github.com/bu-ist/bu-liaison-inquiry/blob/main/src/components/FormBrowser.js#L105

Reply:
@jdub233 Can you take another look? I refactored this to use `async` and try/catch/finally. This works in my sandbox. Is this on track with what you were suggesting?

Reply:
Great, thanks!